cmake/modules/user_cache.cmake

```diff
@@ -0,0 +1,99 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# Copyright (c) 2021, Nordic Semiconductor ASA
+
+# Configure user cache directory.
+#
+# The user cache can be used for caching of data that should be persistent
+# across builds to speed up CMake configure / build system generation and/or
+# compilation.
+#
+# Only data that can be safely re-generated should be placed in this cache.
+#
+# Zephyr build system uses this user cache to store Zephyr compiler check
+# results which significantly improve toolchain testing performance.
+# See https://github.com/zephyrproject-rtos/zephyr/pull/7102 for details.
+#
+# Outcome:
+# The following variables will be defined when this CMake module completes:
+#
+# - USER_CACHE_DIR: User cache directory in use.
+#
+# If the above variable is already set when this CMake module is loaded,
+# then no changes to the variable will happen.
+#
+# CMake module dependencies:
+# - python CMake module.
+
+include_guard(GLOBAL)
+
+# Dependencies of this CMake module.
+include(python)
```

Comment:
This comment is superfluous.

It should be obvious that we include things because we depend on them

Reply:
The same applies elsewhere